src/librustdoc/clean/inline.rs

```diff
@@ -216,6 +216,7 @@ pub(crate) fn build_external_trait(cx: &mut DocContext<'_>, did: DefId) -> clean
     let (generics, supertrait_bounds) = separate_supertrait_bounds(generics);
     let is_auto = cx.tcx.trait_is_auto(did);
     clean::Trait {
+        def_id: did,
```

Comment:
I wonder: if we have the trait `DefId`, I think most (if not all) other fields can be computed on demand, no?

Reply:
Probably? What is the purpose of  clean types anyway? 🤔

Reply:
Types "cleaned up" for rustdoc use. :)

Reply:
It just feels (from an outside perspective) that a lot of this can be replaced by `DefId` newtypes + getters 😅

Reply:
`unsafety` or `is_auto` for example. :)

Well, can be done later on.